src/task_monitor.rs

```diff
@@ -128,13 +128,13 @@ impl TaskMonitor {
         // We could use the second element of the tuple as `mut shutdown_receiver`,
         // but for symmetry's sake we create it for every task with `.subscribe()`
         let (shutdown_sender, _) = broadcast::channel(1);
-        let (wake_up_sender, wake_up_receiver) = mpsc::channel(1);
```

Comment:
I would also be partial to increasing the capacity of this channel, but I see no reason why we should limit the speed at which we insert pending identities into the database.